packages/components/image.tsx

```diff
@@ -1,176 +1,69 @@
-/* eslint-disable jsx-a11y/alt-text */
-
 import * as React from "react";
```

Comment:
Is that type of wildcard import safe for the tree-shaking/unused-code-elimination algorithms?

Maybe we could change it for named imports...

Reply:
I had to double check this because I was not 100% sure myself. It appears that whether you use the namespace import or the named import makes no difference. Both will bring the whole module into your code. Which makes sense, because even if you just import one function from a library, like `import { someFunc } from "library";` there is no guarantee that this function is not relying on some side-effect inside of `"library"`. 

In any case, it's up to the creator of the `"library"` to make sure that it's tree shakeable using the [`sideEffects`](https://webpack.js.org/configuration/optimization/#optimizationsideeffects)  or [`/*#__PURE__*/` annotation](https://webpack.js.org/guides/tree-shaking/#mark-a-function-call-as-side-effect-free).

Everything above only holds true for ES Modules, however. If using CommonJS I think the tree-shaking story is more complicated - I haven't looked into it in detail. React is still not shipping the ES Modules build by the way until at least React 18: https://github.com/facebook/react/issues/11503#issuecomment-662647468

- https://webpack.js.org/guides/tree-shaking

- https://stackoverflow.com/questions/58741044/why-webpack-doesnt-tree-shake-the-lodash-when-using-import-as

- https://github.com/webpack/webpack/issues/2713

Reply:
I see. Thanks for the explanation Michal 🙂